### PR TITLE
Add ValidateName to CompositeError

### LINKED ISSUE #31: Add support for ValidateName to CompositeError
`Validation` contains a helper method `ValidateName` that extends the error message with an additional name that's used to generate better error messages for nested properties.  

With `CompositeError` this isn't possible and results in ambiguous error messages.  For consistency in the API `CompositeError` should be updated to include a `ValidateName` method that recursively calls `ValidateName` on any children.

Open Question:
* Should this happen recursively by calling `ValidateName` on any `CompositeError` objects included in the root `CompositeError`?

### PR DESCRIPTION
**Pull request by veleek**
Added a ValidateName method to CompositeError to mirror the same API in Validation type to enable generated validation logic to alias CompositeErrors the same wait it aliases Validations.

Fixes #31

### CODE REVIEW

**Review by casualjim — APPROVED** [review]

**Comment by veleek** [comment]
I took a closer look at this and I'm not sure that it'll do what I'm expecting it to do.  I turns out that `ValidateName` only sets the name property if it's not already set.  This seems to contradict  the comment which is: `// ValidateName produces an error message name for an aliased property`.

My quick overview of some subset of the code we have that's generated using `go-swagger` will never pass `""` as `Validation.Name` so `ValidateName` essentially does nothing.  Is this expected?

**Comment by veleek** [comment]
Added a change showing how I would expect it to actually work.  Note that the tests have not been updated at the moment so they'll still fail because they're written according to the existing behavior.

**Comment by veleek** [comment]
@casualjim - Where's the right place to discuss something like this to get sign-off?

**Review by casualjim — APPROVED** [review]

**Comment by casualjim** [comment]
This PR is still marked as draft, so I can't review it

**Review by casualjim — APPROVED** [review]

**Comment by veleek** [comment]
Hey @casualjim - I see this was merged to unblock youyuanwu.  I don't remember if I ever updated the tests to pass with this change.  I was waiting for some feedback around if we thought this was a reasonable solution before fixing the tests but forgot to follow up with you to ask again.  

I haven't tried running the tests in a while though, so I'm not aware what the current state is.

**Comment by casualjim** [comment]
ic, we can still update the tests. I merged the change into go-swagger master so I had to bring this one along

**Pull request merged**
